cmd/oras/internal/errors/errors.go

```diff
@@ -30,3 +33,13 @@ func NewErrInvalidReference(ref registry.Reference) error {
 func NewErrInvalidReferenceStr(ref string) error {
 	return fmt.Errorf("%s: invalid image reference, expecting <name:tag|name@digest>", ref)
 }
+
+// IsReferrersIndexDelete checks if err is a referrers index delete error.
+func IsReferrersIndexDelete(err error, logger logrus.FieldLogger, path string) bool {
+	var re *remote.ReferrersError
+	if !errors.As(err, &re) || !re.IsReferrersIndexDelete() {
+		return false
+	}
+	logger.Info("Failed to remove the outdated referrers index: %s@%s. Garbage collection may be required.", path, re.Subject.Digest)
```

Comment:
I would suggest printing the status of `oras push/copy/attach` at the beginning if the new referrers are successfully pushed/copied/attached to the registry. How about `Operation succeed but failed to remove the outdated referrers index: %s@%s. Garbage collection may be required.?

Reply:
changed as suggested